File: harddecode/src/main/java/shine/com/cn/testharddecode/MainActivity.java
package shine.com.cn.testharddecode;

/**
 * @author Kevin
 * @data 2018/9/11
 */
public class MainActivity {
    static {
        System.loadLibrary("shineHardDecode");
    }

    public native String stringFromJNI();

    public native int add(int x, int y);

    public native int startAllNetStreamWork();

    public native int getH264Frame(byte[] resultImage,
                                   int nChannelDec,
                                   int nSleepMs
    );

    //0 ok other stop
    public native int getChannelWorkStatus(int nChannelDec);
}

File: libharddecode/src/main/java/shine/com/cn/testharddecode/MainActivity.kt
package shine.com.cn.testharddecode

/**
 * @author Kevin
 * @data 2018/9/27
 */
class MainActivity {

    external fun stringFromJNI(): String

    external fun add(x: Int, y: Int): Int

    external fun startAllNetStreamWork(): Int

    external fun getH264Frame(resultImage: ByteArray,
                              nChannelDec: Int,
                              nSleepMs: Int
    ): Int

    //0 ok other stop
    external fun getChannelWorkStatus(nChannelDec: Int): Int

    companion object {
        private const val TAG = "ShineHardDecode"

        init {
            System.loadLibrary("shineHardDecode")
        }
    }
}